### PR TITLE
Sbt codeartifact

### PR DESCRIPTION
**Pull request by bbstilson**
https://github.com/allenai/scholar/issues/26603

Things changed in this PR:

- bump sbt version
- publish to codeartifact
- add sbt-release, simplifying the release process
- add 2.13 as a cross publish target, update some collections to be cross compatible

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,15 @@
...
-1. Edit `build.sbt` to remove "-SNAPSHOT" from the current version
-1. Create a pull request if desired or push to master if you are only changing the version
-1. Tag the release `git tag -a vX.Y.Z -m "Release X.Y.Z"` replacing X.Y.Z with the correct version
-1. Push the tag back to origin `git push origin vX.Y.Z`
-1. Release the build on Bintray `sbt +publish` (the "+" is required to cross-compile)
-1. Verify publication [on bintray.com](https://bintray.com/allenai/maven)
-1. Bump the version in `build.sbt` on master (and push!) with X.Y.Z+1-SNAPSHOT (e.g., 2.5.1
--SNAPSHOT after releasing 2.5.0)
-
-If you make a mistake you can rollback the release with `sbt bintrayUnpublish` and retag the
```
Comment (ca16) [inline README.md:24]: I did not know `sbt bintrayUnpublish` existed...

**Review by dirkgr — APPROVED** [review]
Without running it myself, this looks like a great change.

File: build.sbt [hunk 2]
```diff
@@ -1,63 +1,13 @@
...
-)
-
-inConfig(IntegrationTest)(org.scalafmt.sbt.ScalafmtPlugin.scalafmtConfigSettings)
-
-lazy val root = (project in file("."))
-    .aggregate(
-      datastore,
-      cli
-    )
-    .configs(IntegrationTest)
```
Comment (ca16) [inline build.sbt:44]: Is this now covered by 
```
  override def projectConfigurations: Seq[Configuration] = Seq(IntegrationTest)
```
in the GlobalPlugin?

Comment (bbstilson) [reply build.sbt:44]: yeah should be

**Review by ca16 — APPROVED** [review]
LGTM

**Pull request merged**